### ISSUE TITLE
Misalignment in settings screen

### ISSUE THREAD

**Issue by mikkamp** [woocommerce/google-listings-and-ads#2274]
### Describe the bug:

When we view the settings screens the left and right column text isn't aligned.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/11388669/3b440ae4-6fda-458b-8016-4aa6dd34d675)

I'd expect the text in the left section to either remove the top padding so it's aligned with the top of the right section, or alternatively have it aligned with the height of the text on the right section.

### Steps to reproduce:
1. Go to a site that has completed onboarding
2. Browse to the Marketing > Google Listings & Ads > Settings
3. View the misalignment

### Additional details:
This especially becomes obvious as we add additional sections to the settings page, some which might not have text on the right side.
![image](https://github.com/woocommerce/google-listings-and-ads/assets/11388669/e5346f49-1178-40ae-a0a1-fd3ffdefb3d3)

**Comment by puntope** [comment]
Thanks for reporting the issue @mikkamp . Notice this is not happening just in the Setting but in all the `.wcdl-section__header` classes.

**Comment by puntope** [comment]
Seems to come from https://github.com/woocommerce/google-listings-and-ads/commit/df97eb0e681071620d13cd9a0bc05fdee4b84577 ( https://github.com/woocommerce/google-listings-and-ads/pull/1853 )

I'm checking why it's changed and how to fix it

**Comment by mikkamp** [comment]
> Notice this is not happening just in the Setting but in all the .wcdl-section__header classes.

Yes it's the same in onboarding and attributes page as well. Wondering though what's expected as when I look at the original figma files it also has some margin/padding there. However in some places that seems to align better with the first line of text on the right side. Do you know more about what's part is expected to line up?
